### PR TITLE
added tools for widget, piece, and module creation. updated readme. a…

### PR DESCRIPTION
**Pull request by colpanik**
…lso got rid of 0.5 stuff, because we're way over that tbh.

### CODE REVIEW

**Comment by agilbert** [comment]
@colpanik how do you feel about namespacing these new commands? `create:widget` etc.

**Comment by colpanik** [comment]
@agilbert I don't have a strong an opinion on it. I considered it and can see the case, but you could also argue the namespacing pattern applied to tasks in apostrophe modules but not necessarily top-level things.

Happy to change it if other people thing it's a good idea.

**Comment by colpanik** [comment]
@stuartromanek @boutell @bgantick @grdunn @austinstarin any thoughts?

File: README.md [hunk 1]
```diff
@@ -19,9 +19,35 @@ To create a new project with the tool:
...
-If you run the `create` command with the `--install` flag, the command will also `npm install` the dependencies of the project and `apostrophe:reset` the database, resulting in a fully bootstrapped project.
+This will create a local copy of our standard [Apostrophe Boilerplate](https://github.com/punkave/apostrophe-boilerplate).
 
-#### Run command-line tasks
+If you run the `create` command with the `--install` flag, the command will also `npm install` the dependencies for the project and `apostrophe:reset` the database, resulting in a fully bootstrapped project.
+
+#### Create a widget
+To bootstrap the necessary files and basic configuration for a new Apostrophe widget, run the following command from within your Apostrophe project's root directory:
+```bash
+apostrophe createWidget <widgetName>
```
Comment (boutell) [inline README.md]: Love it, but this should be hyphenated. All of our tasks are typically run with hyphenated names. Alex's point that it should maybe be widget:create is a reasonable one too but I don't have a strong opinion. Just don't leave it intercap.

Comment (boutell) [inline README.md]: Include an example so people know what you expect them to type here. The widget name vs. the full module name...

File: README.md [hunk 2]
```diff
@@ -19,9 +19,35 @@ To create a new project with the tool:
...
+```
+
+#### Create a piece
+To bootstrap the necessary files and basic configuration for a new Apostrophe piece type, run the following command from within your Apostrophe project's root directory:
+```bash
+apostrophe createPiece <pieceName>
+```
+
+If you run the `createPiece` command with the `--pages` flag, the command will also set up a corresponding pieces-pages module with your new piece type. Similarly, you can run the `createPiece` command with the `--widgets` flag, which will also set up a corresponding pieces-widgets module along with your new piece type. These flags can be used together or separately.
```
Comment (boutell) [inline README.md]: So shiny, please hyphenate the task names or namespace them per Alex.

File: README.md [hunk 3]
```diff
@@ -19,9 +19,35 @@ To create a new project with the tool:
...
+
+```bash
+apostrophe createPiece <pieceName> --pages --widgets
+```
+
+#### Create an empty Apostrophe module
+To bootstrap the necessary files and basic configuration for a super basic Apostrophe module, run the following command from within your Apostrophe project's root directory:
```
Comment (boutell) [inline README.md]: "A brand-new Apostrophe module that doesn't extend one of the usual suspects like pieces or widgets" might make more sense to people than "super basic".

File: README.md [hunk 4]
```diff
@@ -19,9 +19,35 @@ To create a new project with the tool:
...
+```bash
+apostrophe createModule <moduleName>
```
Comment (boutell) [inline README.md]: Please namespace or hyphenate.

File: README.md [hunk 5]
```diff
@@ -33,8 +59,8 @@ The apostrophe-cli assumes the `apostrophe` namespace when executing tasks. This
 apostrophe <task name>
```
Comment (boutell) [inline README.md]: This should be `<namespace>:<task name>`, right? This might be old text though.

Comment (colpanik) [inline README.md]: "The apostrophe-cli assumes the `apostrophe` namespace when executing tasks" so you could run apostrophe taskname if the task you wanted to run was on the apostrophe-namespace... You would have to <namespace>:<task name> if you wanted to run something on another namespace. I added a point clarifying this

**Pull request merged**